### ISSUE TITLE
With a large number of embedded assets, performance is pretty bad

### ISSUE THREAD

**Issue by lenvanessen** [spicywebau/craft-embedded-assets#136]
For us, it takes about 25 seconds for the assets page to load. Primarily, we're waiting on Curl requests to finish.

Is it a possibility to use caching here? Or perhaps store the information in the db and not in the file system?

We're using an S3 bucket, so it has to be a network request in order to fetch the content.

![Schermafbeelding 2020-06-22 om 10 46 04](https://user-images.githubusercontent.com/23024452/85267905-1930f180-b476-11ea-93ea-16e34e2dc6eb.png)

**Comment by sjcallender** [comment]
@lenvanessen How many assets do you have? We just have 70 and the assets page doesn't load anything. We're getting memory_limit errors that are alarmingly high.

```
[29-Jun-2020 15:43:09 America/New_York] PHP Fatal error:  Allowed memory size of 1073741824 bytes exhausted (tried to allocate 20480 bytes) in /Users/stephencallender/Projects/_sites/local-ch/vendor/opis/closure/src/SerializableClosure.php on line 381
```

**Comment by KatieMFritz** [comment]
Hi @ttempleton! When do you expect to release 2.3.0? Not sure whether we can wait, or we need to make a workaround.

**Comment by ttempleton** [comment]
@KatieMFritz We're planning to have 2.3.0 finished and released by the end of next week.

**Comment by ttempleton** [comment]
Update: I've been looking at the performance of Embedded Assets during the assets page load, and identified an unnecessarily large number of reloads of each embedded asset file's contents.  I've made a change so that the plugin will check for the required embedded asset data within an array of embedded asset data already loaded during a request, before trying to load the file contents.  This has now been released as Embedded Assets 2.2.7.  It isn't a complete solution to this issue, as we do still need to remove remote file requests as much as we possibly can, but it should provide a performance improvement in the meantime.

**Comment by sjcallender** [comment]
@ttempleton The other thing that would be good if you could include is why every time we view an embedded asset's Craft thumbnail/preview the plugin generates a lot of "Updating search indexes" queue jobs. During heavy content-editing periods, the queue gets quite large for us.

Watch: After I refresh the Assets view, the queue fills up.
![Screen Capture on 2020-07-07 at 11-38-27](https://user-images.githubusercontent.com/984311/86806115-8d5ac000-c046-11ea-8e68-3651f662a0e9.gif)

**Comment by ttempleton** [comment]
@sjcallender -- if those are Instagram embeds, Embedded Assets is resaving Instagram embeds to update their `dateModified` as part of a fix for issue #114.  It shouldn't be doing this *every* load of the Assets page, though; only after seven days.  I'll look at what's happening there and fix that for 2.3.0.

Also, an update on the performance issues with Embedded Assets and remote asset storage: I've implemented a solution where copies of the embedded asset JSON files will be saved in the `storage/runtime/assets/embeddedassets` directory and loaded from there.  This is currently located in the [`feature/file-cache`](https://github.com/spicywebau/craft-embedded-assets/tree/feature/file-cache) branch, and I intend to release it by the end of the week, once the above Instagram issue and #137 are fixed.

**Comment by ttempleton** [comment]
Embedded Assets 2.3.0 has just been released.  Note that the first load of a Craft asset index page that contains embedded assets will still take as long as it was taking before the update, as it will need to save the local copies.  Future loads of the page should then be improved.

@sjcallender, the issue with Instagram embeds always being resaved and generating search index update jobs is also fixed in Embedded Asset 2.3.0; it will now wait until seven days after an asset's last check for an expired signature to check again and resave.
